h4l/selection/lepton.py

```diff
@@ -33,67 +34,67 @@ def electron_selection(
+        else:
+            BDT = events.Electron.mvaHZZIso
+
+        # pre-UL WP for Run II (miniAOD branch: Run2_CutBased_BTag16)
+        if self.config_inst.campaign.x("preUL", False):
+            electron_base_mask = (
+                (
+                    (pt <= 10.) & (
+                        (fSCeta < 0.8 & BDT > 0.85216885148) |
+                        (fSCeta >= 0.8 & fSCeta < 1.479 & BDT > 0.82684550976) |
+                        (fSCeta >= 1.479 & BDT > 0.86937630022)
                     )
+                ) |
+                (
+                    (pt > 10.) & (
+                        (fSCeta < 0.8 & BDT > 0.98248928759) |
+                        (fSCeta >= 0.8 & fSCeta < 1.479 & BDT > 0.96919224579) |
+                        (fSCeta >= 1.479 & BDT > 0.79349796445))
                 )
+            )
 
-            else:  # UL WP (miniAOD branch Run2_CutBased_UL)
-                # print("This is not preUL!")
-                # from IPython import embed; embed()
-                return (
-                    (
-                        (pt <= 10.) & (
-                            ((fSCeta < 0.8) & (BDT > 0.9128577458)) |
-                            ((fSCeta >= 0.8) & (fSCeta < 1.479) & (BDT > 0.9056792368)) |
-                            ((fSCeta >= 1.479) & (BDT > 0.9439440575))
-                        )
-                    ) |
-                    (
-                        (pt > 10.) & (
-                            ((fSCeta < 0.8) & (BDT > 0.1559788054)) |
-                            ((fSCeta >= 0.8) & (fSCeta < 1.479) & (BDT > 0.0273863727)) |
-                            ((fSCeta >= 1.479) & (BDT > -0.5532483665))
-                        )
+        else:  # UL WP (miniAOD branch Run2_CutBased_UL)
+            electron_base_mask = (
+                (
+                    (pt <= 10.) & (
+                        ((fSCeta < 0.8) & (BDT > 0.9128577458)) |
+                        ((fSCeta >= 0.8) & (fSCeta < 1.479) & (BDT > 0.9056792368)) |
+                        ((fSCeta >= 1.479) & (BDT > 0.9439440575))
+                    )
+                ) |
+                (
+                    (pt > 10.) & (
+                        ((fSCeta < 0.8) & (BDT > 0.1559788054)) |
+                        ((fSCeta >= 0.8) & (fSCeta < 1.479) & (BDT > 0.0273863727)) |
+                        ((fSCeta >= 1.479) & (BDT > -0.5532483665))
                     )
                 )
-        return ak.ones_like(events.Electron.pt)
-    default_mask = (
+            )
+
+    # mask for selecting electrons
+    electron_mask = (
         (pt > min_pt) &
-        (abs(events.Electron.eta) < 2.5) & 
+        (abs(events.Electron.eta) < 2.5) &
         (events.Electron.dxy < 0.5) &
-        (events.Electron.dz < 1.0) & 
-        (abs(events.Electron.sip3d) < 4) & 
-        return_cuts()
+        (events.Electron.dz < 1.0) &
+        (abs(events.Electron.sip3d) < 4) &
+        electron_base_mask
```

Comment:
OK, I now see that you've opted for just a separate mask, which is also a way we could go about this. Still, imho it would be better to separate this out more clearly, also because we actually don't need to run the ID selections at the loose WP (though this shouldn't affect the run time at all of course since these operations aren't very complicated)

Reply:
Maybe this is a matter of opinion, but I didn't think having a separate function was good practice here, since it anyway has no arguments and would always return the same thing. So I opted to store the decision in a variable instead.

Reply:
as discussed earlier in person, we'll probably switch this around a little. In principle I agree that it's not the nicest style to use a function w/o arguments, so we'll see I guess